### DISCUSSION TITLE
Newbie question - how do I invoke DVR_SCAN from within a python module

### DISCUSSION THREAD

**Discussion by sundeepgoel72** [Breakthrough/DVR-Scan#107] [General]
Hi,

how do I invoke DVR_SCAN from within a python module ?  I mean not via CLI but directly by importing it as a module, passing config parameters and filename from within another python script?

regards,
Sundeep

**Comment by Breakthrough** [comment]
Firstly, while this is certainly possible, note that there is no official API for DVR-Scan, and I do not have any plans for creating one.  If you use the Python API directly, it may break in a future version. You can see how to use the API from the unit tests:
https://github.com/Breakthrough/DVR-Scan/blob/master/tests/test_scan_context.py#L62

Note that the API is different than the CLI, and currently uses a bunch of `set_` functions to set various config options before calling `scan_motion`.  The `ScanContext` is the main class you use to set all configuration options and to do the actual scanning.  Take a look at `run_dvr_scan` in `dvr_scan.cli` to see how all CLI options are passed to `ScanContext`:
https://github.com/Breakthrough/DVR-Scan/blob/master/dvr_scan/cli/controller.py#L180

If you want, I can add the ability to pass command-line arguments to `run_dvr_scan` in the next version if you want to file a feature request.  That is probably best if you want to also load a config file from Python.  Like the API, this may also break if command line options change, but this would be far easier to maintain when it does need to be changed.

E.g. ideally, you could do something like:

```python
from dvr_scan.cli import run_dvr_scan
run_dvr_scan(['-i', 'video.mp4', '-so'])    # Returns 0 on success, error code otherwise.
```

Lastly, you can also consider just using `subprocess` to invoke the `dvr-scan` command inside a Python script.

Hope this helps!
